File: PHP/0007.php
class Solution {

    /**
     * @param Integer $x
     * @return Integer
     */
    function reverse($x) {
        $rev = 0;
        $INT_MAX = 2147483647; // Max value for a 32-bit signed integer
        $INT_MIN = -2147483648; // Min value for a 32-bit signed integer
        
        while ($x != 0) {
            $pop = $x % 10;
            $x = (int)($x / 10); // Use (int) to truncate towards zero
            
            // Check for overflow/underflow before multiplying rev by 10 and adding pop
            if ($rev > ($INT_MAX - $pop) / 10 || $rev < ($INT_MIN - $pop) / 10) {
                return 0;
            }
            
            $rev = $rev * 10 + $pop;
        }
        
        return $rev;
    }
}

File: Rust/0007.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let mut rev = 0;
        let mut num = x;
        
        while num != 0 {
            let pop = num % 10;
            num /= 10;
            
            // Check for overflow/underflow before multiplying rev by 10 and adding pop
            if rev > i32::MAX / 10 || (rev == i32::MAX / 10 && pop > 7) {
                return 0;
            }
            if rev < i32::MIN / 10 || (rev == i32::MIN / 10 && pop < -8) {
                return 0;
            }
            
            rev = rev * 10 + pop;
        }
        
        rev
    }
}
